src/components/chip-set/chip-set.tsx

```diff
@@ -653,89 +575,82 @@ export class ChipSet {
-                    >
-                        {chip.text}
-                    </span>
-                </span>
-                {this.renderBadge(chip)}
-            </div>
-        );
-    }
-
     private renderInputChip(chip: Chip, index: number) {
+        const chipProps = this.getChipProps(chip, 'default');
+
         return [
-            <div
+            <limel-chip
+                key={chip.id}
                 class={{
-                    'mdc-chip': true,
-                    'mdc-chip--selected': this.inputChipIndexSelected === index,
-                    disabled: this.disabled,
+                    'can-be-removed': this.inputChipIndexSelected === index,
                 }}
-                role="row"
-                id={`${chip.id}`}
-                onClick={this.catchInputChipClicks}
-            >
-                {chip.icon ? this.renderChipIcon(chip) : null}
-                {this.renderChipLabel(chip)}
-                {this.renderChipRemoveButton(chip)}
-            </div>,
+                {...chipProps}
+                onRemove={this.removeChip(chip.id)}
+            />,
             this.renderDelimiter(),
         ];
     }
 
-    private catchInputChipClicks(event) {
-        event.stopPropagation();
+    private removeChip = (id: string | number) => () => {
+        const newValue = this.value.filter((chip) => {
+            return `${chip.id}` !== `${id}`;
+        });
+        this.change.emit(newValue);
+    };
+
+    private getChipProps(chip: Chip, chipType: ChipType) {
+        const removable =
+            this.type === 'input' &&
+            (chip.removable || !this.readonly || !this.disabled);
+
+        return {
+            role: 'row',
+            id: `${chip.id}`,
+            text: chip.text,
+            icon: chip.icon,
+            badge: chip.badge,
+            selected: chip.selected,
+            disabled: this.disabled,
+            readonly: this.readonly,
+            type: chipType,
+            removable: removable,
+            onClick: this.catchInputChipClicks(chip),
+            ...(chip.href && {
+                link: {
+                    href: getHref(chip.href),
+                    target: getTarget(chip.href),
+                },
+            }),
+        };
     }
 
-    private renderChipIcon(chip: Chip) {
-        const name = getIconName(chip.icon);
-        const color = getIconColor(chip.icon, chip.iconFillColor);
-        const backgroundColor = getIconBackgroundColor(
-            chip.icon,
-            chip.iconBackgroundColor,
-        );
-        const title = getIconTitle(chip.icon, chip.iconTitle);
-        const style = {};
-        if (color) {
-            style['--icon-color'] = color;
+    private catchInputChipClicks = (chip: Chip) => () => {
+        if (this.type !== 'input') {
+            const selected = chip.selected;
+            chip.selected = !selected;
+            this.selectedChip = chip;
+            this.updateSelectedChipIds();
```

Comment:
Here is the background story: 
I think previously mdc chipset's own scripts took care of a lot of things. We weren't sure how exactly they worked. Sometimes simply adding the class of `mdc-chip` on `limel-chip` made it work like before. But sometimes it didn't help. So we decided to remove everything that comes from MDC, and make sure that all examples and implementations work like before, but with our own logics instead of MDC's. Specially because we wanna get rid of MDC generally.

Now I cannot judge the implementation, as the King Bef has done it and my knowledge is too little for evaluating it. But as far as our tests showed, things worked like before when we were done.

But if things can be improved, just feel free to do so with fresh eyes.

Reply:
> The way the code currently looks, it seems a little unnecessary to have the this.selectedChip property at all, especially as a @State-prop, since all it does is hold the value of the selected chip from line 631, to line 632, where it is immediately deconstructed, and only the id is used 😅

The reason is when selecting(or highlighting or clicking whatever we call it 😅) the chips having the chip type = 'filter' for some reason it could not trigger re-rendering even if `this.selectedChipIds` is changed. You can remove the  state `this.selectedChip` and try it. I am interested to know why it could not re-render updating `this.selectedChipIds`

Reply:
> I am interested to know why it could not re-render updating `this.selectedChipIds`

That's because the `splice` on line 650 and the `push` on line 652 mutates the existing array, instead of replacing it with a new array. Because it's the same array, only with changed content, Stencil has no way of detecting the change.

Reply:
Nice I did not know that Stencil doesn't detect the modification. It doesn't detect the changes since the array reference remains the same. 

I will change the `splice` code with

```
this.selectedChipIds = this.selectedChipIds
                .filter((chipId: number | string) => chipId !== id);
```

and the `push` one to `this.selectedChipIds = [...this.selectedChipIds, id];`.
Finally I will remove `this.selectedChip`